### ISSUE TITLE
Create a GitHub repository template that would enable easier onboarding to generator and writing templates

### ISSUE THREAD

**Issue by derberg** [asyncapi/generator#402]
#### Reason/Context
The generator has already many different functionalities that if utilized, can make the writing of the template much easier. Writing of first template can be overwhelming especially that we do not have a single template at the moment that would reach 1.0 release and use all generator features. We need good docs and examples, but best would be to have both in a GitHub repo template so creating own repo for own template would be super easy (something like https://github.com/jsynowiec/node-typescript-boilerplate)

#### Description
- create a new repo that will be a template
- it should have ready to use out-of-the-box release pipeline with GitHub Actions configured for conventional commits spec
- sample template could be a simple [Mermaid](https://github.com/mermaid-js/mermaid) class diagram generation for the `components/schemas` to visualize application data model
- in package.json all config options should be used
- it should use our generic libraries with hooks and filters
- there should be some custom hook, maybe the generation of pdf and image with the diagram using puppeteer https://github.com/puppeteer/puppeteer and this could be customized with parameters
- show as many Nunjucks possibilities as possible, with partials as mandatory to showcase reusability
- there should be some custom filter to show how you can make template writing easier and limit the size of the template (filter is Nunjucks functionality that we enable for others to provide and register for the generation)
- find a good reason to utilize file templates like `$$channel$$` 🤔 
- write good docs to fully describe the entire template and all functionalities
- have unit tests and even snapshot tests for the HTML
- maybe possible to do some Docker thingy too? 🤔 

goal is to get a best practices template

**Comment by fmvilas** [comment]
I have the feeling this template is already too complicated for a starter use case. Have you considered creating multiple templates? For instance:

1. A simple one. Only the bare minimum to create a template. A hello world.
1. Another one demoing some non-basic features.
1. Another one demoing some advanced features.

The last two could be just one as a starting point. And, in every case, I think this template should have no purpose at all. What would be the difference with cloning the Node.js or HTML templates, for someone who's starting?

**Comment by derberg** [comment]
maybe not complicated but rather complex. Mermaid is in scope just to have more fun from writing a template. It will be very simplistic and its only purpose is learning, not using in production (on a contrary to Node.js and HTML templates) so it will just touch features to present them. It will be followed by docs, of course, describing every part of the template, even scripts from NPM so nothing will be left for interpretation. I will explain what parts can be removed if someone doesn't want to have a release pipeline for example. 

So basically the purpose will be some html and mermaid + some stuff that will be needed if I won't find a way o present some feature with mermaid (like partials or files templates).

I think that once the template is ready/final and I will be done with writing docs for it we can have another look. It might be that after doing it I will realize it is better to now maybe record a video or something that explains the template step by step, or create a set of tutorials for it.
